### PR TITLE
EQL: Remove support for `=` for comparisons

### LINKED ISSUE #62650: EQL:  Remove (was Deprecate) `=` in favor of `==` for equality checks
A topic that was raised in https://github.com/elastic/elasticsearch/issues/61883#issuecomment-690281936 is deprecating usage of `=` when doing equality checks.
EQL does use `=` for assignments so using it for equality checks instead of `==` is potentially confusing (is it an assignment or an equality check?).
It would be good to get some numbers on the usage to see how often this occurs.
Based on that we can decide whether to go with a safe route (deprecation-only with removal later down the road) or straight removal.

### PR DESCRIPTION
**Pull request by matriv**
Since `=` is rarely used and is undocumented we its support for
equality comparisons keeping `==` as the only option. `=` is now only
used for assignements like in `maxspan=10m`.

Closes: #62650

### CODE REVIEW

**Comment by matriv** [comment]
/cc @jrodewig I've checked the docs, (we don't document the `=` for comparisons) but it would be great if you could double check if any other change is needed. Thx!

@costin Should we mark it as breaking after all?

**Review by costin — APPROVED** [review]
LGTM

File: docs/reference/eql/functions.asciidoc [hunk 1]
```diff
@@ -1086,7 +1086,7 @@ expressions.
 process.name == "*regsvr32*" or process.name == "*explorer*"
 wildcard(process.name, "*regsvr32*", "*explorer*")
 
-// process.name = "regsvr32.exe"
+// process.name == "regsvr32.exe"
```
Comment (jrodewig) [inline docs/reference/eql/functions.asciidoc:1089]: This is actually an assignment.
```suggestion
// process.name = "regsvr32.exe"
```

**Review by jrodewig — APPROVED** [review]
LGTM.

**Review by astefan — APPROVED** [review]
LGTM

**Comment by matriv** [comment]
@elasticmachine run elasticsearch-ci/1
@elasticmachine run elasticsearch-ci/bwc
@elasticmachine run elasticsearch-ci/default-distro

**Comment by costin** [comment]
Let's err on the safe side and mark it as such. In the unlikely case folks are using it, they'll want to know about it even for an experimental release.

**Comment by matriv** [comment]
@costin also agree with marking it "breaking".

**Pull request merged**